### ISSUE TITLE
.html URL rewriting issues

### ISSUE THREAD

**Issue by niconii** [neocities/neocities#427]
I don't know to what extent you read Twitter replies these days, and I thought they might go unnoticed so I'll collect some of the complaints here:

https://twitter.com/JonathanWojcik/status/1609043416897425408
> This actually makes 20 years worth of disqus comments unviewable, which were an integrated part of my site's content. Disqus considers it a different, new page without the .html and therefore starts a new thread. The comment community is already in a panic over it

https://twitter.com/JonathanWojcik/status/1609047078080237568
> If there's a way for this to simply not automatically redirect, and leave the .html in the browser's url bar, then everything's fine, but otherwise this wipes out a community of over a million posts on our end

https://twitter.com/JonathanWojcik/status/1609051395172478977
> (3/3) there is at least a possibility of migrating the disqus comments, so we'll be looking into that, but I'm sure disqus isn't the only thing affected?

https://twitter.com/cylindricities/status/1608955477412708352
> Slight problem:
> (I made a twitter account just for this occasion)
> No?? You may not know it, but webriddles (like mine), (ex. http://polychromasia.neocities.org) rely on .html filenames and extensions for levels.
> Please make this optional for every webpage, but make it naturally (/example).

https://twitter.com/cylindricities/status/1609378249146089472
> Happy New Year to everyone who lost the mental and emotional battle of keeping .htm extensions on your pages to a cat on the internet

The breakages are one aspect of this, but I think it's also worth noting that people may have deliberately chosen to use `foo.html`, `foo.htm`, or `foo` for aesthetic reasons, and this change has essentially removed two of those options for the sake of making `foo` slightly more convenient. Add on top of that the way it complicates moving to other static web hosts, and to me personally the tradeoffs don't make sense.

Personally, I would recommend reverting this change. The sooner the better, before many new pages rely on this URL rewriting, and before people try drastic things to try and fix things like Disqus comments.

**Comment by prino** [comment]
This needless change also affects ahrefs statistics, with suddenly lots of redirect warnings...

What is the use of it?

**Comment by necrophcodr** [comment]
Yeah this is not a change I'd say I agree with either. In general I'm not too fond of things being done implicitly, when there's an explicit way of being even more clear. Having `/example` and `/example/` NOT be the same might also be confusing even further.

**Comment by lime360** [comment]
it may be off-topic but i made a joke:

glitch being owned by a random company: *chill*

neocities adding cleaner links: ***NOT CHILL***

**Comment by flamedfury** [comment]
> Yeah this is not a change I'd say I agree with either. In general I'm not too fond of things being done implicitly, when there's an explicit way of being even more clear. Having `/example` and `/example/` NOT be the same might also be confusing even further.

This has just screwed up a bunch of my links after moving to Neocities. My previous host would treat `/example` and `/example/` as the same, they would resolve to the same page. Now any internal or external link that links to `/example/` 404's! I also don't see any way that we are able to 301 redirect `/example/` to `/example` , or am I missing something?

**Comment by erisdev** [comment]
This is currently blocking me migrating my static site to Neocities. Surely part of the appeal of this service is that folks can easily set up fun little amateur personal pages? Y'all named yourselves after GeoCities after all. We're not here to make sleek professional looking web sites. If I really wanted "clean" extension-less paths I would just stick an `index.html` in a folder with the name I wanted, which most static site generators are already doing anyways. I had to actually fight 11ty to make it not do that.

**Comment by somefan0102** [comment]
i'm having a similar problem too and it's slightly frustrating.
can't move `/music/index.html` to `/music.html` without getting a bad gateway

**Comment by jodo13** [comment]
Come on...

I should have read your blog before register with you https://blog.neocities.org/blog/2022/12/30/cleaner-links

I'm having issues with geocities.ws and I read recommendations that neocities.org was just a basic static web files host but... now with rewrites???

No, thank you, no.

Who cares if new generations can't type .html, or .htm, or because major browsers "hide" the actual URL in the url bar to look cleaner.

Extensions represent something. I don't want cleaner URLs. Who wants them? I don't.

Regretting :-(

**Comment by shan995-cell** [comment]
How is this still not an opt-out feature after almost 4 years?
